vumi/tests/helpers.py

```diff
@@ -1454,7 +1454,12 @@ def cleanup(self):
                 unclosed_managers.append(manager)
             if purge:
                 try:
-                    yield self._purge_riak(manager)
+                    # Create a new client for cleanup
+                    purge_manager = self.get_riak_manager(config={
+                        'bucket_prefix': manager.bucket_prefix
+                    })
```

Comment:
Not unless we're talking to multiple Riak clusters something. In that case, we'd need more extensive changes to the test infrastructure anyway.